lib-compose/src/main/java/com/what3words/components/compose/maps/W3WMapManager.kt

```diff
@@ -574,6 +593,45 @@ class W3WMapManager(
             )
         }
     }
+
+    /**
+     * Handle zoom option for a [W3WCoordinates] with multiple zoom options which will use the zoom level
+     * if it's provided or the default zoom level.
+     */
+    @UiThread
```

Comment:
The @UiThread annotation in Android is a contract or indicator, not a guarantee and it does not enforce it at runtime or switch to the main thread automatically. To ensure thread safety and compliance with coroutine rules, please modify the function to explicitly run on the Dispatchers.Main dispatcher